// mySQLconnect/mySQLconnect.js

//mySQLconnect Module


//exports.queryTest= queryTest;

// Create a connection to MySql Server and Database
var connection = require('mysql').createConnection({
  host : 'localhost',
  user : 'admin',
  password : '1234'
});

var TB_CHAT = 'simpoze.cht_chat';
var TB_USERS = 'simpoze.tbusers';
var SELECT_CHT = "SELECT * from "+TB_CHAT;
var INSERT_CHT = "INSERT INTO " +TB_CHAT+ " (username, text) VALUES ";
var INSERT_USER = "INSERT INTO " +TB_USERS+ " (username, password) VALUES ";
var SELECT_USER = "SELECT * from " +TB_USERS+ " WHERE (USERNAME =@username AND PASSWORD = @password)";


function querySelectChat(socket){
	var query = SELECT_CHT;
	console.log(query);
    connection.query(query, function(err, rows){
        if(err != null) { 
            socket.emit("DB", "Error in Query: " + err);
        }else{// Shows the results:
			for (var i=0; i<rows.length; i++){
				 console.log(rows[i]);
				 socket.emit("DBnews",rows[i]);
			}
        }
    });
}

function queryInsertChat(socket, data){
	var query = INSERT_CHT;
	if (!query){ //ERROR IN QUERY
			console.log("Error in Query! ");
			return; 
		}
	query += "('"+ data.username +"','" + data.message +"')";
	console.log(query);
    connection.query(query, function(err, rows){
        if(err != null) {
            socket.emit("DBnews", "Error in Query: " + err);
        }
    });
}

function querySelectUser(data){
	var query = SELECT_USER;
	var res;
	
	console.log("DATA: " + data + data.username + data.password);
	query = query.replace('@username', "'"+data.username+"'");
	query = query.replace('@password', "'"+data.password+"'");
	console.log(query);
    
	connection.query(query, function(err, rows){
        if(err != null) { 
            console.log("DB", "Error in Query: " + err);
            return "ERROR";
        }else{// Shows the results:
        	//result = console.log(rows[0]);
			 //socket.emit("DBnews",rows[i]);
        	//{username:rows[0].username, password:rows[0].password};
        	res = rows[0];
        	return res;
        }
    });
	
	return res;
}

exports.queryInsertChat = queryInsertChat;
exports.querySelectChat = querySelectChat;
exports.querySelectUser = querySelectUser;
exports.connection = connection;


//function queryTest(){
//	var query = SELECT_CHT;
//	connection.query(query, function(err, rows){
//        if(err != null) { 
//        
//        }else{// Shows the results:
//			for (var i=0; i<rows.length; i++){
//				 console.log(rows[i]);
//			}
//        }
//	});
//}
